README.md

```diff
@@ -33,6 +39,60 @@ export default defineConfig({
 });
 ```
 
+## Using Loader
+
+The plugin comes with support for loading `.res` files directly. This is optional and in most cases not necessary,
+but can be useful in combination with ["in-source": false](https://rescript-lang.org/docs/manual/latest/build-configuration#package-specs).
+When using this setting without the loader, it becomes impossible to import local files using relative paths.
+Take for example the following code:
+
+```res
+%%raw(`import "./local.css"`)
+@module("./local.js") external runSomeJs: unit => unit = "default"
+```
+
+The bundler will fail when reaching this file, since the imports are resolved relative to the generated JS file (which resides in `lib`),
+but the `.css` and `.js` files are not copied into this directory. By using the loader it no longer fails since the bundler will resolve
+the files relative to the `.res` file instead.
+
+### Configuration
+
+When using Vite in library mode, just use the `.res` as entry point:
+
+```js
+import { defineConfig } from 'vite';
+import createReScriptPlugin from '@jihchi/vite-plugin-rescript';
+
+export default defineConfig({
+  plugins: [createReScriptPlugin()],
+  build: {
+    lib: {
+      entry: 'src/App.res',
+    },
+  },
+});
+```
+
+For HTML entry points, it must be imported using inline JS:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite App</title>
+  </head>
+  <body>
+    <div id="root"></div>
+    <script type="module">
+      import '/src/App.res';
```

Comment:
**Question:** What is in the `App.res`, something like this?

```res
ReactDOM.render(
  <React.StrictMode> <XXXXXX /> </React.StrictMode>,
  ReactDOM.querySelector("#root")->Belt.Option.getExn,
)
```

I've tried it with [jihchi/vitejs-template-react-rescript](https://github.com/jihchi/vitejs-template-react-rescript), it works with `import /src/Main.res;`, `App.res` is a react component in that template repository.

Reply:
> **Question:** What is in the `App.res`, something like this?
> 
> ```rescript
> ReactDOM.render(
>   <React.StrictMode> <XXXXXX /> </React.StrictMode>,
>   ReactDOM.querySelector("#root")->Belt.Option.getExn,
> )
> ```
> 
> I've tried it with [jihchi/vitejs-template-react-rescript](https://github.com/jihchi/vitejs-template-react-rescript), it works with `import /src/Main.res;`, `App.res` is a react component in that template repository.

Yes exactly, it would be something like that. The `App.res` was just an example, I will change it to `Main.res` to better reflect the structure in your starter template.